app/src/components/ShareLinkButton.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { Tooltip, TooltipTrigger } from "react-aria-components";
+import { useLocation } from "react-router";
+
+import { Button, Icon, Icons, Text, View } from "@phoenix/components";
+import { useNotifySuccess } from "@phoenix/contexts";
+
+export const ShareLinkButton = ({
+  successText,
+  tooltipText = "Copy link to clipboard",
+  preserveSearchParams = false,
+}: {
+  successText?: string;
+  tooltipText?: string;
+  preserveSearchParams?: boolean;
+}) => {
+  const location = useLocation();
+  const notifySuccess = useNotifySuccess();
+  return (
+    <TooltipTrigger delay={200}>
+      <Button
+        size="S"
+        variant="quiet"
+        leadingVisual={<Icon svg={<Icons.ExternalLinkOutline />} />}
+        onPress={() => {
+          const url = new URL(location.pathname, window.location.origin);
+          if (preserveSearchParams) {
+            url.search = location.search;
+          }
+          navigator.clipboard.writeText(url.toString());
+          notifySuccess({
+            title: successText ?? "Link copied to clipboard",
+            expireMs: 1000,
+          });
+        }}
+      />
+      <Tooltip offset={10}>
```

Comment:
placement wise we might just want to put it where all the other controls for the span is.